### ISSUE TITLE
Documentation: make clear when updates are performed if a tag is unchanged but the image changes

### ISSUE THREAD

**Issue by micw** [keel-hq/keel#297]
With docker, tags are mutable: the referenced image may change for any tag (not only for latest). For example even if a project is using semantic versioning for the project, a new docker image could be available if the base image changes without changing the project's version (most common case: security update of the base image).

The documentation says for policies:

   force: force update even if tag is not semver, ie: latest

The example comments this with:

    # force policy will ensure that deployment is updated
    # even when tag is unchanged (latest remains)
    keel.sh/policy: force

Does this mean that only with "force", updates are performed if the tag is not changed but the image is? This should be made clear in the doc.

IMO it would make sense to change the policy name from "force" to "keep" and enable/disable updating on unchanged tags separately and not implicitly.

**Comment by micw** [comment]
After the reading of the sources, I think this is the current behaviour - can you confirm this?

Semantic (All, Major, Minor, Patch): Updates are only performed if the tag increases. If an image is pushed to the existing tag, this is simply ignored

Force (with matchTag): Updates are only performed if an image is pushed to the existing tag.

Force (without matchTag): Updates performed on any tag push. So when 2.0 is installed and someone pushes to 1.0, a downgrade would be performed.

**Comment by rusenask** [comment]
you are correct :)

**Comment by Astha-Sharma** [comment]
@rusenask Force (without matchTag) does not seem to work for me when I install 1.0-1 and a tag with 1.0-2 is pushed. Could you help me understand what policy/configuration would be best to use if I want an update to happen in this case?

**Comment by rusenask** [comment]
Hi, force will work when you are using GCR pubsub or webhooks. I guess you are using polling to trigger updates?

**Comment by Astha-Sharma** [comment]
You are right, I am using polling on AWS ECR for updates with helm provider. But using force without matchTag seems to work like using force with matchTag for my case. I didn't find in documentation that force wasn't for polling.
It would still be great to know if I can perform upgrade from 1.0-1 to 1.0-2 (what I would normally expect force without matchTag to do).

**Comment by rusenask** [comment]
Polling for multiple tags can't work with non semver policy (like force or regex) because we don't really know which tag is higher/newer than the previous.

Would it be difficult for you to switch to semver versioning scheme? It looks like you are almost there :)

**Comment by Astha-Sharma** [comment]
Thanks @rusenask :)
It's a bit difficult for me to shift to semver versioning right now but I'll try to hack a bit into keel's code to see if I can modify it for my usecase!

**Comment by dardanbekteshi** [comment]
> Polling for multiple tags can't work with non semver policy (like force or regex) because we don't really know which tag is higher/newer than the previous.
> 
> Would it be difficult for you to switch to semver versioning scheme? It looks like you are almost there :)

I couldn't find this anywhere in the documentation. Maybe it should be added? @rusenask
